### PR TITLE
Spot fixes

### PR DESCRIPTION
**Pull request by ehowe**

### CODE REVIEW

File: lib/fog/aws/requests/compute/request_spot_instances.rb [hunk 1]
```diff
@@ -71,7 +71,7 @@ def request_spot_instances(image_id, instance_type, spot_price, options = {})
             options.merge!(Fog::AWS.indexed_param('LaunchSpecification.SecurityGroupId', [*security_group_ids]))
           end
           if options['LaunchSpecification.UserData']
-            options['LaunchSpecification.UserData'] = Base64.encode64(options['LaunchSpecification.UserData'])
+            options['LaunchSpecification.UserData'] = Base64.encode64(options['LaunchSpecification.UserData']).chomp!
```
Comment (lanej) [inline lib/fog/aws/requests/compute/request_spot_instances.rb]: The in-place part of `chomp!` is not likely not necessary.

File: tests/requests/compute/spot_instance_tests.rb [hunk 2]
```diff
@@ -42,10 +42,8 @@
       data
     end
 
-    pending if Fog.mocking?
```
Comment (lanej) [inline tests/requests/compute/spot_instance_tests.rb]: love it when this happens.

File: tests/requests/compute/spot_instance_tests.rb [hunk 3]
```diff
@@ -42,10 +42,8 @@
...
-
     tests("#describe_spot_instance_requests").formats(@spot_instance_requests_format) do
-      Fog::Compute[:aws].describe_spot_instance_requests('spot-instance-request-id' => [@spot_instance_request_id]).body
+      data = Fog::Compute[:aws].describe_spot_instance_requests('spot-instance-request-id' => [@spot_instance_request_id]).body
```
Comment (lanej) [inline tests/requests/compute/spot_instance_tests.rb]: what purpose does the `data` assignment serve?

File: lib/fog/aws/requests/compute/request_spot_instances.rb (hunk 1 above)
Comment (ehowe) [inline lib/fog/aws/requests/compute/request_spot_instances.rb]: It is.  the API expects the output of `strict_encode64`, but this is how this happens everywhere else, I assume for 1.8 compatibility.

File: tests/requests/compute/spot_instance_tests.rb (hunk 3 above)
Comment (ehowe) [inline tests/requests/compute/spot_instance_tests.rb]: Nothing, side effect of testing that I forgot to remove before commiting

File: lib/fog/aws/requests/compute/request_spot_instances.rb (hunk 1 above)
Comment (lanej) [inline lib/fog/aws/requests/compute/request_spot_instances.rb]: Interesting.  I can't get 1.8 to install any more so I'm going to take your word for it.

Comment (ehowe) [inline lib/fog/aws/requests/compute/request_spot_instances.rb]: Yeah, the API explodes with a generic "Invalid Request" if you don't `chomp!` the \n off of the userdata.  It's neat.  Took me forever to figure out.

**Comment by geemus** [comment]
Thanks!

**Pull request merged**